### ISSUE TITLE
Locale-dependent decimal parsing issues

### ISSUE THREAD

**Issue by GoaLitiuM** [mikke89/RmlUi#505]
The library does not seem to parse some attributes correctly when the global C++ locale is changed to locale which uses commas as decimal separators instead of periods. For example the `<progress>` elements are not rendered correctly with non-standard locale in our engine:
```cpp
<progress class="gauge" direction="clockwise" start-edge="bottom" value="0.5" max="1"/>
<progress class="horizontal" value="0.55" max="1.0"/>
<progress class="vertical" direction="top" value="0.5" max="1"/>
```
![image](https://github.com/mikke89/RmlUi/assets/4064397/94aee1b3-025b-42fd-a970-83b4372c75e2)

When I wrap the `RmlContext::Render()` calls with these, the `<progress>` elements starts rendering properly:
```cpp
std::locale oldLocale = std::locale::global(std::locale::classic());
context->Render();
std::locale::global(oldLocale);
```
![image](https://github.com/mikke89/RmlUi/assets/4064397/de3465ea-05f0-4ef1-b50c-c8b278f8be06)

This same issue also happens with stylesheets containing decimal values, so I already worked around that issue by changing the locale while loading the document, but I can't justify doing the same here while issuing rendering calls every frame to the library.

**Comment by mikke89** [comment]
You're right, currently we are using locale-aware conversion functions, pretty much for all our conversions. I mean, I'd love to use `std::to_chars` and `std::from_chars` instead, but that is C++17 which we don't use yet, and even with that, support is very limited even in recent std library versions.

I would generally discourage ever changing the global C or C++ locale. It's just asking for trouble in my view. Instead, pass the desired locale to the specific functions that you want to be locale-dependent.
